### PR TITLE
[Send] Add/Edit functionality

### PR DESCRIPTION
**Pull request by vincentsalucci**
## Code Changes
- **jslib**: Updated to 8a3b551
- **messages.json**: Fixed `Link vs link` debacle. Added new strings.
- **routing-animations**: Added animation routes for both adding and editing Sends (to/from)
- **app-routing.module.ts**: Added `add-send` and `edit-send` to routing paths
- **app.component.ts**: Added scope cleanup for `vault/GroupingsComponentScope`.  Removed unnecessary `TODO`
- **app.module.ts**: Added `SendAddEditComponent`. Added `DatePipe` provider that is used within the new component.
- **box.scss**: Added `box-header-exapandble` and `flex-label` stylings. Adjusted `input` to not include radio styling. Added `radio-group` styling.
- **misc.scss**: Added `clickable` class styling to add hover states for `app-callout`
- **send-add-edit.component.html/ts**: Created new component.
- **send-groupings.component.ts**: Removed unused import/constructor init. Added routing to `add/edit` component.
- **send-type.component.ts**: Added routing to `add/edit` component.

## Notes
- **File/Text selector**: ~~The extension doesn't use radio buttons currently, so for the sake of existing patterns the `SendType` selector is a dropdown menu. Let me know if we still want to change it to radio buttons (or some variation of that).~~
- **File Warning callout**: Firefox/Safari have issues with opening finder windows/dismissing popup focus so I've used an `app-callout` banner to alert the user (under certain circumstances) that they need to pop out the window in order to choose a file. I didn't want to automatically pop the window out (like we will do in the next release for Attachments) because the user could simply change the `SendType`. I also didn't want to use a dialog for the same reason. The UX is much smoother with the warning banner and voluntary click. 
- **Strings**: I've shortened a few strings (those used with right justified checkboxes) for a nicer look (no wrapping)

## Screenshots
14-add-file-collapsed
15-add-text-collapsed
16-add-text-expanded
17-add-text-expanded
18-edit-text-collapsed
19-edit-text-expanded
20-edit-text-expanded
21-firefox-file-warning

## Testing Considerations
- `Add` send from both `groupings` and `type` components
- `Edit` send from both `groupings` and `type` components
- `Add File` send works from all browser platforms (chromium, firefox, safari)
- `Delete` existing send from `Edit` component
- All options are functioning properly

### CODE REVIEW

**Comment by cscharf** [comment]
> File/Text selector: The extension doesn't use radio buttons currently, so for the sake of existing patterns the SendType selector is a dropdown menu. Let me know if we still want to change it to radio buttons (or some variation of that).

Yes, we want to be consistent with the rest of the clients on this. Please use radio-buttons @vincentsalucci

File: src/popup/send/send-add-edit.component.html [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+            <div class="box-footer">
+                {{'sendNameDesc' | i18n}}
+            </div>
+        </div>
+        <!-- Type Options -->
+        <div class="box" *ngIf="!editMode">
+            <div class="box-content">
+                <div class="box-content-row" appBoxRow>
+                    <label for="sendTypeOptions">{{'sendTypeHeader' | i18n}}</label>
+                    <select id="sendTypeOptions" name="SendTypeOptions" [(ngModel)]="send.type"
```
Comment (cscharf) [inline src/popup/send/send-add-edit.component.html:40]: As noted, let's try to get these into a consistent look/feel radio buttons vs. drop down.

File: src/popup/send/send-add-edit.component.html [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+        <div class="right">
+            <button type="submit" appBlurClick [disabled]="form.loading">
+                <span [hidden]="form.loading">{{'save' | i18n}}</span>
+                <i class="fa fa-spinner fa-lg fa-spin" [hidden]="!form.loading" aria-hidden="true"></i>
+            </button>
+        </div>
+    </header>
+    <content *ngIf="send">
+        <!-- File Warning -->
+        <app-callout type="warning" icon="fa-external-link" title="{{'sendFileCalloutHeader' | i18n}}"
```
Comment (cscharf) [inline src/popup/send/send-add-edit.component.html:18]: Good call on this, I like it.

Comment (vincentsalucci) [reply src/popup/send/send-add-edit.component.html]: Thank you! Updated the UI a bit to match where we use the pop out icon elsewhere.

File: src/popup/send/send-add-edit.component.html (hunk 1 above)
Comment (vincentsalucci) [reply src/popup/send/send-add-edit.component.html]: Updated!

**Review by cscharf — APPROVED** [review]
I think that all looks good

**Pull request merged**